### ISSUE TITLE
Delete applied to private name via optional chain

### ISSUE THREAD

**Issue by mgaudet** [tc39/test262#2665]
I think a test case for  `delete o?.#x` needs to be added.

**Comment by rwaldron** [comment]
@leobalter does this fall in scope for the refactoring you wanted to do?

**Comment by leobalter** [comment]
This is not covered.

I did some work but missed covering optional chain as [UnaryExpressions](https://tc39.es/ecma262/#sec-unary-operators).

Funny enough, `delete {}?.x` is valid.

The version with private fields is a SyntaxError regardless of the delete.

```js
// valid
class C {
  #x = 42;
  m() {
    var o = this;
    return o.#x;
  }
}

// SyntaxError
class C {
  #x = 42;
  m() {
    var o = this;
    return o?.#x;
  }
}
```

We might coverage for private fields + optional chain, but needs verification.

**Comment by mgaudet** [comment]
I don't think that second example is a syntax error since https://github.com/tc39/proposal-class-fields/pull/301

**Comment by leobalter** [comment]
Thanks! I was not up to date with the private fields proposal and then we just need to verify coverage.

@mgaudet can you help me pointing to the part where `delete o?.x` would be forbidden?

**Comment by leobalter** [comment]
I found it: https://tc39.es/proposal-class-fields/#sec-static-semantics-early-errors

**Comment by caiolima** [comment]
IIRC, we had tests to cover that `o?.#x` was a `SyntaxError` and this got reverted into https://github.com/tc39/test262/pull/2566. I don't remember seeing or adding any test that covers `delete o?.#x` though.
